vision/airdrop_vision_pipeline.py

```diff
@@ -0,0 +1,148 @@
+"""Runs vision code to find judges during an airdrop"""
+
+from typing import Callable
+
+import time
+import cv2
+
+import vision.common.constants as consts
+
+from vision.common.bounding_box import BoundingBox
+from vision.emergent_object.emergent_object import create_emergent_model
+from vision.emergent_object.emergent_object_processing import calc_emerg_obj_min_dist
+
+import vision.pipeline.emergent_pipeline as emg_obj
+import vision.pipeline.pipeline_utils as pipe_utils
+
+# The furthest a judge can move between images in feet
+MAX_JUDGE_MOVEMENT_F: float = 15
+
+
+def airdrop_pipeline(camera_data_path: str, state_path: str, output_path: str) -> None:
+    """
+    Finds all judges in the images in the camera_data_path json file.
+
+    Unlike with the Flyover Vision Pipeline, the output_path json will be updated continuously.
+
+    Parameters
+    ----------
+    camera_data_path: str
+        The path to the json file containing the CameraParameters entries
+    state_path: str
+        A text file containing True if all images have been taken and False otherwise
+    output_path: str
+        The json file name and path to save the data in
+    """
+
+    # Load model
+    emg_model: Callable[[consts.Image], str] = create_emergent_model()
+
+    # List of filenames for images already completed to prevent repeating work
+    completed_images: list[str] = []
+
+    # The data from the previous run
+    prev_data: list[BoundingBox] = []
+
+    # True only for the first time an image is processed - prevents errors from prev_data
+    first_detection: bool = True
+
+    # Wait for and process unfinished images until no more images are being taken
+    all_images_taken: bool = False
+    while not all_images_taken:
+        # Wait to check the files instead of spamming it
+        time.sleep(0.1)
+
+        # Check if all images have been taken
+        all_images_taken = pipe_utils.flyover_finished(state_path)
+
+        # Load in the json containing the camera data
+        image_parameters: dict[str, consts.CameraParameters] = pipe_utils.read_parameter_json(
+            camera_data_path
+        )
+
+        # Loop through all images in the json - if it hasn't been processed, process it
+        for image_path in image_parameters.keys():
+            if image_path not in completed_images:
+                # Save the image path as completed so it isn't processed again
+                completed_images.append(image_path)
+
+                # Load the image to process
+                image: consts.Image = cv2.imread(image_path)
+
+                # Get the camera parameters from the loaded parameter file
+                camera_parameters: consts.CameraParameters = image_parameters[image_path]
+
+                # Find potential judges in the image
+                current_data = emg_obj.find_humanoids(
+                    image, emg_model, camera_parameters, image_path
+                )
```

Comment:
Instead of `current_data`, which isn't very clear, you could use `current_humanoids`. Same thing with `prev_data`